lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1502,6 +1508,45 @@ def _node_generate_min_capability(
         min_cap: schema.NodeSpace = req_cap.generate_min_capability(azure_cap_copy)
         return min_cap
 
+    def _generate_data_disks(
+        self,
+        node: Node,
+        azure_node_runbook: AzureNodeSchema,
+    ) -> None:
+        marketplace = self.get_image_info(
+            azure_node_runbook.location, azure_node_runbook.marketplace
+        )
+        # some images has data disks by default
+        # e.g. microsoft-ads linux-data-science-vm linuxdsvm 21.05.27
+        # we have to inject below part when dataDisks section added in
+        #  arm template,
+        # otherwise will see below exception:
+        # deployment failed: InvalidParameter: StorageProfile.dataDisks.lun
+        #  does not have required value(s) for image specified in
+        #  storage profile.
+        for default_data_disk in marketplace.data_disk_images:
+            self.data_disks.append(
+                DataDiskSchema(
+                    node.capability.data_disk_caching_type,
+                    default_data_disk.additional_properties["sizeInGb"],
+                    azure_node_runbook.disk_id,
+                    "FromImage",
+                )
+            )
+        assert isinstance(node.capability.data_disk_count, int)
+        disk_count: int = 0
+        while disk_count < node.capability.data_disk_count:
```

Comment:
Sorry, haven't notice it in first time review. It can be, `for disk_index in range(node.capability.data_disk_count):`. It doesn't need to operate the disk_count variable.

Reply:
thanks, fixed